base/loading.jl

```diff
@@ -302,7 +302,22 @@ function find_package(arg)
     return locate_package(pkg)
 end
 
-## package identity: given a package name and a context, try to return its identity ##
+"""
+    Base.identify_package(name::String)::Union{PkgId, Nothing}
+    Base.identify_package(where::Union{Module,PkgId}, name::String)::Union{PkgId, Nothing}
+
+Identify the package by its name from the current environment stack, returning
+its `PkgId`, or `nothing` if it cannot be found.
+
+If only the `name` argument is provided, it searches each environment in the
+stack and its named direct dependencies.
+
+There `where` argument provides the context from where to search for the
+package: in this case it first checks if the name matches the context itself,
+otherwise it searches all recursive dependencies (from the resolved manifest)
+until it locates the context `where`, and from there resolves the name from its
```

Comment:
"resolves the name" is kind of unclear, perhaps "looks up the UUID" or something?